### PR TITLE
IBM Dynamical Decoupling Tutorial

### PR DESCRIPTION
**Pull request by knsmith**
IBM Dynamical decoupling tutorial that builds on the introductory material in #464

### CODE REVIEW

File: docs/source/optimizations/ibm/ibmq_dd.ipynb [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ee5db276-d951-4fe0-b5f3-194ba3c9f865",
+   "metadata": {},
+   "source": [
+    "[Dynamical decoupling (DD)](https://arxiv.org/abs/quant-ph/9803057) is an error mitigation technique used to suppress noise added to computation from system-environment interaction (i.e. decoherence error). DD is implemented during algorithm execution by adding special gate sequences that are equal to the identity operation in periods of qubit idling. Examples of DD sequences include XX, XY4, XY8, and UDD, among others.\n",
+    "\n",
+    "Below is a brief tutorial on SuperstaQ dynamical decoupling optimizations for the IBM family of superconducting quantum devices. In the example, a CPMG sequence, XX, will be used. \n",
```
Comment (dowusu-antwi) [inline docs/source/optimizations/ibm/ibmq_dd.ipynb:31]:
```suggestion
    "Below is a brief tutorial on Superstaq dynamical decoupling optimizations for the IBM family of superconducting quantum devices. In the example, a CPMG sequence, XX, will be used. \n",
```

Comment (knsmith) [reply docs/source/optimizations/ibm/ibmq_dd.ipynb]: Good catch - thanks!!

File: docs/source/optimizations/ibm/ibmq_dd.ipynb [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    "# Optional imports\n",
+    "import os  # used if setting a token as an environment variable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cf7190af-16a0-4dc7-a5b3-c893c950b8b2",
+   "metadata": {},
+   "source": [
+    "To interface Superstaq via Qiskit, we must first instantiate a provider in `qiskit-superstaq` with `SuperstaQProvider()`. We then supply a Superstaq API token (or key) by either providing the API token as an argument of `qss.SuperstaQProvider()` or by setting it as an environment variable (see more details [here](https://docs-superstaq.readthedocs.io/en/stable/get_started/basics/basics_qss.html#Import-Superstaq-and-requirements))."
```
Comment (dowusu-antwi) [inline docs/source/optimizations/ibm/ibmq_dd.ipynb:67]:
```suggestion
    "To interface Superstaq via Qiskit, we must first instantiate a provider in `qiskit-superstaq` with `SuperstaqProvider()`. We then supply a Superstaq API token (or key) by either providing the API token as an argument of `qss.SuperstaqProvider()` or by setting it as an environment variable (see more details [here](https://docs-superstaq.readthedocs.io/en/stable/get_started/basics/basics_qss.html#Import-Superstaq-and-requirements))."
```

File: docs/source/optimizations/ibm/ibmq_dd.ipynb [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "dfa13fec-7580-4d74-9c67-b997fe690b5e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the qiskit superstaq provider for Superstaq backend\n",
+    "# Provide your api key to the using the \"api_key\" argument if\n",
+    "# SUPERSTAQ_API_KEY environment variable is not set.\n",
+    "provider = qss.SuperstaQProvider()"
```
Comment (dowusu-antwi) [inline docs/source/optimizations/ibm/ibmq_dd.ipynb:80]:
```suggestion
    "provider = qss.SuperstaqProvider()"
```

File: docs/source/optimizations/ibm/ibmq_dd.ipynb [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+       "                                                                                                                                                                2  3  1  0 "
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc_compiled.circuit.draw(fold=-1)"
```
Comment (dowusu-antwi) [inline docs/source/optimizations/ibm/ibmq_dd.ipynb:206]: micro-nit: I usually add `qc.draw()` to the same cell defining/generating it, just to be more compact

File: docs/source/optimizations/ibm/ibmq_dd.ipynb [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+       "                                                                                                                                                                                              2  3  1  0 "
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc_compiled_dd.circuit.draw(fold=-1)"
```
Comment (dowusu-antwi) [inline docs/source/optimizations/ibm/ibmq_dd.ipynb:294]: (ditto)

File: docs/source/optimizations/ibm/ibmq_dd.ipynb (hunk 2 above)
Comment (knsmith) [reply docs/source/optimizations/ibm/ibmq_dd.ipynb:67]: Good catch! thanks :)

File: docs/source/optimizations/ibm/ibmq_dd.ipynb (hunk 4 above)
Comment (knsmith) [reply docs/source/optimizations/ibm/ibmq_dd.ipynb:197]: That's fair! Went ahead and made the change.

**Review by dowusu-antwi — APPROVED** [review]
Nice, lgtm!

One last micro-nit: I would comment out the first cell and then re-run everything (then uncomment again), so we get the nice numerical progression, lol
image

**Comment by knsmith** [comment]
> Nice, lgtm!
> 
> One last micro-nit: I would comment out the first cell and then re-run everything (then uncomment again), so we get the nice numerical progression, lol image

Done!

**Pull request merged**